sources/metrics/detectors/failover/FailoverDetector.swift

```diff
@@ -0,0 +1,45 @@
+//  Copyright 2019, Oath Inc.
+//  Licensed under the terms of the MIT License. See LICENSE.md file in project root for terms.
+import Foundation
+import PlayerCore
+
+extension Detectors {
+    final class Failover {
+        
+        private enum AdResult {
+            case other
+            case processing
+            case failover
+            case noAds
+        }
+        
+        private var adSessionIDs = Set<UUID>()
+        private var currentResult = AdResult.other
+        
+        func process(vrmResponse: VRMResponse?,
+                     currentGroup: VRMCore.Group?,
+                     groupQueue: [VRMCore.Group],
+                     adSessionID: UUID?) -> Bool {
+            guard let vrmResponse = vrmResponse,
+                let adSessionID = adSessionID,
+                adSessionIDs.contains(adSessionID) == false else {
+                    currentResult = .other
+                    return false
+            }
+            guard vrmResponse.groups.isEmpty == false,
+               currentResult != .noAds else {
+                currentResult = .noAds
+                adSessionIDs.insert(adSessionID)
+                return false
+            }
+            guard currentGroup == nil,
+                groupQueue.isEmpty else {
+                    currentResult = .processing
+                    return false
+            }
+            currentResult = .failover
+            adSessionIDs.insert(adSessionID)
```

Comment:
`currentResult = smth and insert of ID` can be moved to separate function and reused.